tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/dataprovider/DataProviderManager.java

```diff
@@ -222,10 +224,23 @@ public List<IDataProviderDescriptor> getAvailableProviders(@Nullable ITmfTrace t
             return Collections.emptyList();
         }
         List<IDataProviderDescriptor> list = new ArrayList<>();
+        TmfAnalysisOutputManager manager = TmfAnalysisOutputManager.getInstance();
```

Comment:
This introduces a coupling of analysis modules and data providers. Data Providers don't need to use analysis modules. However, in most case so far there is analysis module defined that the data provider is using. But, if we look outside the code coupling, data providers are outputs and hence it probably ok to expand the hideOutput extension point to data providers, especially, since analysis_id is optional in the extension point.

Reply:
It's not adding dependencies on modules however it causes the data provider manager to need to consider analysisId inside the outputId. We're only accessing the TmfAnalysisOutputManager to reuse its reading of the hideOutput extension point attribute. This could be factored out, however it would still be tied to the 'org.eclipse.linuxtools.tmf.core.analysis' extension point.

Perhaps it would be better to replicate the functionality of 'hideOutput' in 'org.eclipse.tracecompass.tmf.core.dataprovider' extension point? Then we can remove the handling of analysisId completely (it could still be part of the outputId string but that would be opaque to the implementation).

Reply:
Sounds reasonable to have it in the data provider extension point. Would you implement a regex pattern to be able hide all data providers with for example the same customization?